packages/core/utils/lib/async.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+function pipeAsync(...methods) {
+  return async (data) => {
+    let res = data;
+
+    for (const method of methods) {
+      res = await method(res);
+    }
+
+    return res;
+  };
+}
+
+/**
+ * Map function callback.
+ * @callback mapAsyncCallback
+ * @param {*} value
+ * @param {number} index
+ */
+/**
+ * Async iteration over an array of promises
+ * @param {promise<*>[]} promiseArray
+ * @returns {function(callback: mapAsyncCallback): promise<*[]>}
+ */
+function mapAsync(promiseArray) {
+  /**
+   * @param {mapAsyncCallback} callback
+   * @returns promise<*[]>
+   */
+  return (callback) => {
```

Comment:
We could use a more meaningful name here as usually callback are mostly for executing something when processing is done.

Reply:
I agree, I have used the "callback" naming by default, but I didn't find the right word for those functions. Maybe `iteratorFunc` but it doesn't feel right either.

Reply:
A lot of libs use `iteratee` for the iteration func. I'm not again a simple `mapFunc|mapper` too